sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -1471,3 +1481,186 @@ def test_gaussian_mixture_all_init_does_not_estimate_gaussian_parameters(
     # The initial gaussian parameters are not estimated. They are estimated for every
     # m_step.
     assert mock.call_count == gm.n_iter_
+
+
+@pytest.mark.parametrize("init_params", ["random", "random_from_data"])
+@pytest.mark.parametrize("covariance_type", ["full", "tied", "diag", "spherical"])
+@pytest.mark.parametrize(
+    "array_namespace, device_, dtype",
+    yield_namespace_device_dtype_combinations(),
+    ids=_get_namespace_device_dtype_ids,
+)
+def test_gaussian_mixture_array_api_compliance(
+    init_params, covariance_type, array_namespace, device_, dtype, global_random_seed
+):
+    """Test that array api works in GaussianMixture.fit()."""
+    X, _ = make_blobs(
```

Comment:
I think this test is not using the dtype parameter to test different dtypes

Reply:
That's a good catch indeed, we noticed it with @StefanieSenger last week and for some reason the test fail with float32, which we need to investigate in more details.